modules/custom/az_select_menu/src/Plugin/Block/AzSelectMenu.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+namespace Drupal\az_select_menu\Plugin\Block;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Template\Attribute;
+use Drupal\menu_block\Plugin\Block\MenuBlock;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides an extended Menu block.
+ *
+ * @Block(
+ *   id = "az_select_menu",
+ *   admin_label = @Translation("Quickstart select menu block"),
+ *   category = @Translation("Quickstart select menu block"),
+ *   deriver = "Drupal\menu_block\Plugin\Derivative\MenuBlock",
+ *   forms = {
+ *     "settings_tray" = "\Drupal\system\Form\SystemMenuOffCanvasForm",
+ *   },
+ * )
+ */
+class AzSelectMenu extends MenuBlock {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    $instance = parent::create($container, $configuration, $plugin_id, $plugin_definition);
+    $instance->menuParentFormSelector = $container->get('menu.parent_form_selector');
+    $instance->entityTypeManager = $container->get('entity_type.manager');
+    return $instance;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockForm($form, FormStateInterface $form_state) {
+    $config = $this->configuration;
+    $defaults = $this->defaultConfiguration();
+
+    $form = parent::blockForm($form, $form_state);
+    $form['menu_levels']['depth']['#attributes'] = ['disabled' => 'disabled'];
+    $form['menu_levels']['depth']['#default_value'] = 1;
+    $form['menu_levels']['depth']['#description'] = 'This maximum number includes the initial level. This has to be set to 1 for Quickstart Select Menus.';
+    $form['menu_levels']['expand_all_items']['#default_value'] = 0;
+    $form['menu_levels']['expand_all_items']['#attributes'] = ['disabled' => 'disabled'];
+    $form['menu_levels']['expand_all_items']['#states']['visible'] = [
+      ':select[name="settings[depth]"]' => ['!value' => 1],
+    ];
```

Comment:
Could we just set `#access` to `FALSE` for both of these?
```suggestion
    $form['menu_levels']['depth']['#access'] = FALSE;
    $form['menu_levels']['expand_all_items']['#access'] = FALSE;
```